File: bloodsmearimageanalysisproject/project/server/test_mongodb_connection.js
/**
 * MongoDB Connection Test Script (Node.js)
 * Tests the connection to MongoDB and verifies database setup
 */

const { MongoClient } = require('mongodb');

// MongoDB Configuration (Authenticated)
const MONGODB_URI = 'mongodb://bhanu:bhanu123@localhost:27017/bloodsmear?authSource=admin';
const DB_NAME = 'bloodsmear';

async function testMongoDBConnection() {
    console.log('='.repeat(60));
    console.log('MongoDB Connection Test (Node.js)');
    console.log('='.repeat(60));

    let client;

    try {
        // Attempt to connect to MongoDB
        console.log(`\n1. Connecting to MongoDB at: ${MONGODB_URI}`);
        client = new MongoClient(MONGODB_URI, {
            useUnifiedTopology: true,
            serverSelectionTimeoutMS: 5000
        });

        await client.connect();
        console.log('   ✓ Successfully connected to MongoDB!');

        // Access the database
        console.log(`\n2. Accessing database: ${DB_NAME}`);
        const db = client.db(DB_NAME);
        console.log('   ✓ Database accessed successfully');

        // List existing collections
        const collections = await db.listCollections().toArray();
        const collectionNames = collections.map(c => c.name);
        console.log(`   Existing collections: ${collectionNames.length > 0 ? collectionNames.join(', ') : 'None (database is empty)'}`);

        // Verify required collections
        console.log('\n3. Verifying required collections...');
        const requiredCollections = ['users', 'analyses', 'results'];

        for (const collectionName of requiredCollections) {
            if (!collectionNames.includes(collectionName)) {
                await db.createCollection(collectionName);
                console.log(`   ✓ Created collection: ${collectionName}`);
            } else {
                console.log(`   ✓ Collection exists: ${collectionName}`);
            }
        }

        // Test write operation
        console.log('\n4. Testing write operation...');
        const testCollection = db.collection('test_connection');
        const testDoc = { test: 'connection', status: 'success', timestamp: new Date() };
        const insertResult = await testCollection.insertOne(testDoc);
        console.log(`   ✓ Write test successful (ID: ${insertResult.insertedId})`);

        // Test read operation
        console.log('5. Testing read operation...');
        const retrievedDoc = await testCollection.findOne({ _id: insertResult.insertedId });
        console.log(`   ✓ Read test successful:`, retrievedDoc);

        // Clean up test document
        await testCollection.deleteOne({ _id: insertResult.insertedId });
        console.log('   ✓ Cleanup completed');

        // Display database stats
        console.log('\n6. Database Statistics:');
        for (const collectionName of requiredCollections) {
            const count = await db.collection(collectionName).countDocuments({});
            console.log(`   - ${collectionName}: ${count} documents`);
        }

        console.log('\n' + '='.repeat(60));
        console.log('✓ All tests passed! MongoDB is ready to use.');
        console.log('='.repeat(60));

        return true;

    } catch (error) {
        console.error('\n✗ ERROR:', error.message);
        console.log('\nMake sure MongoDB is running on localhost:27017');
        console.log('\nTo start MongoDB:');
        console.log('1. Open MongoDB Compass and connect to mongodb://localhost:27017/');
        console.log('2. Or run: net start MongoDB (requires admin)');
        return false;

    } finally {
        if (client) {
            await client.close();
            console.log('\nConnection closed.');
        }
    }
}

// Run the test
testMongoDBConnection()
    .then(success => {
        process.exit(success ? 0 : 1);
    })
    .catch(error => {
        console.error('Unexpected error:', error);
        process.exit(1);
    });

File: bloodsmearimageanalysisproject/project/test_mongodb_connection.py
"""
MongoDB Connection Test Script
Tests the connection to MongoDB and verifies database setup
"""

from pymongo import MongoClient
from pymongo.errors import ConnectionFailure, ServerSelectionTimeoutError
import sys

# MongoDB Configuration (Authenticated)
MONGODB_URI = 'mongodb://bhanu:bhanu123@localhost:27017/bloodsmear?authSource=admin'
DB_NAME = 'bloodsmear'

def test_mongodb_connection():
    """Test MongoDB connection and database setup"""
    print("=" * 60)
    print("MongoDB Connection Test")
    print("=" * 60)
    
    try:
        # Attempt to connect to MongoDB
        print(f"\n1. Connecting to MongoDB at: {MONGODB_URI}")
        client = MongoClient(MONGODB_URI, serverSelectionTimeoutMS=5000)
        
        # Test the connection
        print("2. Testing connection...")
        client.admin.command('ping')
        print("   ✓ Successfully connected to MongoDB!")
        
        # Access the database
        print(f"\n3. Accessing database: {DB_NAME}")
        db = client[DB_NAME]
        
        # List existing collections
        collections = db.list_collection_names()
        print(f"   ✓ Database accessed successfully")
        print(f"   Existing collections: {collections if collections else 'None (database is empty)'}")
        
        # Create collections if they don't exist
        print("\n4. Verifying required collections...")
        required_collections = ['users', 'analyses', 'results']
        
        for collection_name in required_collections:
            if collection_name not in collections:
                db.create_collection(collection_name)
                print(f"   ✓ Created collection: {collection_name}")
            else:
                print(f"   ✓ Collection exists: {collection_name}")
        
        # Test write operation
        print("\n5. Testing write operation...")
        test_collection = db['test_connection']
        test_doc = {'test': 'connection', 'status': 'success'}
        result = test_collection.insert_one(test_doc)
        print(f"   ✓ Write test successful (ID: {result.inserted_id})")
        
        # Test read operation
        print("6. Testing read operation...")
        retrieved_doc = test_collection.find_one({'_id': result.inserted_id})
        print(f"   ✓ Read test successful: {retrieved_doc}")
        
        # Clean up test document
        test_collection.delete_one({'_id': result.inserted_id})
        print("   ✓ Cleanup completed")
        
        # Display database stats
        print("\n7. Database Statistics:")
        for collection_name in required_collections:
            count = db[collection_name].count_documents({})
            print(f"   - {collection_name}: {count} documents")
        
        print("\n" + "=" * 60)
        print("✓ All tests passed! MongoDB is ready to use.")
        print("=" * 60)
        
        return True
        
    except ConnectionFailure:
        print("\n✗ ERROR: Could not connect to MongoDB")
        print("  Make sure MongoDB is running on localhost:27017")
        print("\n  To start MongoDB:")
        print("  1. Open MongoDB Compass and connect to mongodb://localhost:27017/")
        print("  2. Or run: net start MongoDB (requires admin)")
        return False
        
    except ServerSelectionTimeoutError:
        print("\n✗ ERROR: MongoDB server is not responding")
        print("  Make sure MongoDB service is running")
        print("\n  To start MongoDB:")
        print("  1. Open MongoDB Compass and connect to mongodb://localhost:27017/")
        print("  2. Or run: net start MongoDB (requires admin)")
        return False
        
    except Exception as e:
        print(f"\n✗ ERROR: {str(e)}")
        return False
    
    finally:
        try:
            client.close()
            print("\nConnection closed.")
        except:
            pass

if __name__ == "__main__":
    success = test_mongodb_connection()
    sys.exit(0 if success else 1)
